### PR TITLE
Flp 4691 code splitting

### PR DESCRIPTION
**Pull request by thoughtbyte**

### CODE REVIEW

File: config/webpack.defaults.js [hunk 1]
```diff
@@ -14,17 +14,46 @@ const cssOptions = {
...
+    '@babel/plugin-syntax-dynamic-import',
+    '@babel/plugin-syntax-import-meta',
+    '@babel/plugin-proposal-class-properties',
+    '@babel/plugin-proposal-json-strings',
+    [
+      '@babel/plugin-proposal-decorators',
+      {
+        legacy: true,
+      },
+    ],
```
Comment (carlyle) [inline config/webpack.defaults.js]: This plugin may need to be loaded before `@babel/plugin-proposal-class-properties` is per https://babeljs.io/docs/en/babel-plugin-proposal-decorators#note-compatibility-with-babel-plugin-proposal-class-properties

File: config/webpack.defaults.js [hunk 2]
```diff
@@ -14,17 +14,46 @@ const cssOptions = {
...
+    '@babel/plugin-syntax-dynamic-import',
+    '@babel/plugin-syntax-import-meta',
+    '@babel/plugin-proposal-class-properties',
+    '@babel/plugin-proposal-json-strings',
+    [
+      '@babel/plugin-proposal-decorators',
+      {
+        legacy: true,
```
Comment (carlyle) [inline config/webpack.defaults.js]: Would it be possible to use this plugin in standard mode instead of legacy mode?

Comment (carlyle) [reply config/webpack.defaults.js]: It also looks like `@babel/plugin-proposal-class-properties` may need to be run in 'loose' mode if we're using legacy decorators per https://babeljs.io/docs/en/babel-plugin-proposal-decorators#note-compatibility-with-babel-plugin-proposal-class-properties

File: template/.babelrc [hunk 3]
```diff
@@ -1,5 +1,36 @@
...
+    "@babel/plugin-syntax-dynamic-import",
+    "@babel/plugin-syntax-import-meta",
+    "@babel/plugin-proposal-class-properties",
+    "@babel/plugin-proposal-json-strings",
+    [
+      "@babel/plugin-proposal-decorators",
+      {
+        "legacy": true
+      }
+    ],
```
Comment (carlyle) [inline template/.babelrc]: See comments on `config/webpack.defaults.js` above

File: config/webpack.defaults.js (hunk 1 above)
Comment (thoughtbyte) [reply config/webpack.defaults.js]: fixed!

**Pull request merged**